src/scenes/Event/components/EventTxHistory/resultHistory.js

```diff
@@ -19,10 +19,31 @@ export default class EventResultHistory extends Component {
     oracles: PropTypes.array.isRequired,
   };
 
+  checkType(oracle, index) {
+    if (oracle.token === Token.Qtum) {
+      return <FormattedMessage id="str.betRound" defaultMessage="Betting Round" />;
+    } else if (index === 1) {
+      return <FormattedMessage id="str.resultSetRound" defaultMessage="Result Setting Round" />;
+    }
+    return <FormattedMessage id="str.voteRound" defaultMessage="Voting Round {idx}" values={{ idx: index - 1 }} />;
+  }
+
   render() {
     const { classes, oracles } = this.props;
-
-    const sortedOracles = _.orderBy(oracles, ['endTime']);
+    let sortedOracles = _.orderBy(oracles, ['endTime']);
+    if (sortedOracles.length) {
+      const row1 = sortedOracles.splice(0, 1)[0];
+      const rowLeft = sortedOracles;
+      const resultSettingOracle = {};
+      resultSettingOracle.endTime = rowLeft[0].endTime;
+      resultSettingOracle.token = rowLeft[0].token;
+      resultSettingOracle.resultIdx = row1.resultIdx;
+      resultSettingOracle.options = row1.options;
+      resultSettingOracle.amounts = row1.amounts;
+      resultSettingOracle.amounts.fill(0);
+      resultSettingOracle.amounts[resultSettingOracle.resultIdx] = 100;
+      sortedOracles = [row1, resultSettingOracle, ...rowLeft];
```

Comment:
instead of reconstructing the entire array, how about inserting the `Result Setting Round` oracle in position 1?